rs/ledger_suite/icp/ledger/src/lib.rs

```diff
@@ -504,3 +622,114 @@ pub fn change_notification_state(
         TimeStamp::from(now()),
     )
 }
+
+pub fn is_ready() -> bool {
+    LEDGER_STATE.with(|s| matches!(*s.borrow(), LedgerState::Ready))
+}
+
+pub fn panic_if_not_ready() {
```

Comment:
In the ICRC-1 ledger, this function is also invoked at the beginning of public canister entry points, e.g., https://github.com/dfinity/ic/blob/313268652881c097abd0b34cc89a6131ec500abf/rs/ledger_suite/icrc1/ledger/src/main.rs#L664C5-L664C26

Reply:
Good point, added!